deploy/baremetal-operator_servicemonitor.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: baremetal-operator
```

Comment:
We should make sure this name is consistent with the name for the component we're using during installation. Talk to @sadasu about what name we settled on.

Reply:
Currently, we are using the name "metal3-baremetalhost-controller". This code is under development and this name could change during review.